mops/src/utils/first_occurences.cu

```diff
@@ -0,0 +1,113 @@
+#include "mops/cuda_first_occurences.hpp"
+#include "mops/cuda_utils.cuh"
+
+#include <cstdint>
+#include <cstdio>
+#include <cuda.h>
+#include <cuda_runtime.h>
+#include <memory>
+
+#define NEIGHBOUR_NEDGES_PER_BLOCK 512
```

Comment:
any chance of renaming the variables to no longer refer to nodes/edges? Sice the operations themself are not defined in these terms.

Reply:
fixed this for first_occurences, still need to apply it to OPSA_cu